python-checks/src/main/java/org/sonar/python/checks/ArgumentNumberCheck.java

```diff
@@ -55,19 +57,54 @@ public class ArgumentNumberCheck extends PythonSubscriptionCheck {
   public void initialize(Context context) {
     context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, ctx -> {
       CallExpression callExpression = (CallExpression) ctx.syntaxNode();
-      Symbol symbol = callExpression.calleeSymbol();
-      if (symbol != null && symbol.kind() == Symbol.Kind.FUNCTION) {
-        FunctionSymbol functionSymbol = (FunctionSymbol) symbol;
-        if (isException(callExpression, functionSymbol)) {
-          return;
-        }
-        checkPositionalParameters(ctx, callExpression, functionSymbol);
-        checkKeywordArguments(ctx, callExpression, functionSymbol, callExpression.callee());
-      }
 
+      getFunctionSymbol(callExpression)
+        .ifPresent(functionSymbol -> checkFunctionSymbol(ctx, callExpression, functionSymbol));
     });
   }
 
+  private static Optional<FunctionSymbol> getFunctionSymbol(CallExpression callExpression) {
+    Symbol calleeSymbol = callExpression.calleeSymbol();
+    if (calleeSymbol == null) {
+      return Optional.empty();
+    }
+
+    if (calleeSymbol.is(Symbol.Kind.FUNCTION)) {
+      return Optional.of(calleeSymbol).map(FunctionSymbol.class::cast);
+    }
+
+    if (calleeSymbol.is(Symbol.Kind.AMBIGUOUS)) {
+      var ambiguousSymbol = (AmbiguousSymbol) calleeSymbol;
+
+      var functionSymbols = ambiguousSymbol.alternatives()
+        .stream()
+        .filter(symbol -> symbol.is(Symbol.Kind.FUNCTION))
+        .map(FunctionSymbol.class::cast)
+        .collect(Collectors.toList());
+
+      var isEqualAmountOfArguments = functionSymbols.stream()
+        .map(FunctionSymbol::parameters)
+        .filter(Objects::nonNull)
+        .map(List::size)
+        .distinct()
+        .count() == 1;
+
+      if (isEqualAmountOfArguments) {
```

Comment:
Argument count may be the same, but what if they have different keywords? 
I guess we could then raise a false positive due to that.
Maybe the simplest approach would be to make sure they also have the same keywords and also positional/keyword-only properties. (and we probably need some tests for that as well)